File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visit;
    static Queue<Integer> queue = new LinkedList<>();
    
    private static void bfs(int i, int[][] computers, int n) {
        queue.offer(i);
        visit[i] = true;
        while (!queue.isEmpty()) {
            int value = queue.poll();
            for(int j=0; j<n; j++) {
                if(visit[j] == false && computers[value][j] == 1) {
                    visit[j] = true;
                    queue.offer(j);
                }
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visit = new boolean[n];
        
        for(int i=0; i<n; i++) {
            if(visit[i] == false) {
                bfs(i, computers, n);
                answer += 1;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    
    def bfs(i):
        queue = deque([i])
        visited[i] = True  # 방문 처리

        while queue:
            x = queue.popleft()
            for j in range(n):  # 모든 노드를 확인
                if computers[x][j] == 1 and not visited[j]:  # 연결되어 있고 방문하지 않았다면
                    visited[j] = True
                    queue.append(j)

    for i in range(n):
        if not visited[i]:  # 방문하지 않은 노드라면 새로운 네트워크 발견
            bfs(i)
            answer += 1  # 새로운 네트워크 개수 증가
    
    return answer
